### PR TITLE
Display external storage files on portrait and landscape screens

### PR DESCRIPTION
**Pull request by ecaterinagherbovei**
So, in this pull request, I introduced the viewModel ```AllFilesViewModel```, where I have ```fetchFiles()``` function. In order to extract files and folders from external memory I decided to use following method:
```Kotlin
val path = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            System.getenv("EXTERNAL_STORAGE")
        } else {
            Environment.getExternalStorageDirectory().absolutePath
        }
``` 
```System.getenv("EXTERNAL_STORAGE")``` return mentioned environment variable, in my case it returns ```\sdcard``` path, where are stored all external files. I tested this function for more devices and it works correctly. 
240719378_341255051076256_63901845552030623_n

And the same for landscape layout 
240807160_385462866263463_1050309583185200137_n

### CODE REVIEW

File: app/src/main/java/com/example/filemanager/AllFilesFragment.kt [hunk 1]
```diff
@@ -50,26 +40,18 @@ class AllFilesFragment : Fragment(), ItemClickListener {
                 RecyclerView.VERTICAL
             )
         )
+        val view = binding.root
+        return view
```
Comment (gdragan) [inline app/src/main/java/com/example/filemanager/AllFilesFragment.kt]: the view variable seems to be redundant, it is not used. Can simply return the `binding.root`

File: app/src/main/java/com/example/filemanager/AllFilesViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import android.content.Context
+import android.os.Build
+import android.os.Environment
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import java.io.File
+
+class AllFilesViewModel(
+    @SuppressLint("StaticFieldLeak")
+    private val context: Context
```
Comment (gdragan) [inline app/src/main/java/com/example/filemanager/AllFilesViewModel.kt]: I don't see where this context is used in the class? Maybe you can remove it ?

File: app/src/main/java/com/example/filemanager/AllFilesViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    val files = MutableLiveData<List<ListModel>>()
+
+    fun fetchFiles() {
+        val path = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            System.getenv("EXTERNAL_STORAGE")
+        } else {
+            Environment.getExternalStorageDirectory().absolutePath
+        }
+        //Log.d("ENV_VARS", System.getenv().toString())
```
Comment (gdragan) [inline app/src/main/java/com/example/filemanager/AllFilesViewModel.kt]: Do you need these logs?, If they are not used then you can remove them I guess?

File: app/src/main/java/com/example/filemanager/ListAdapter.kt [hunk 4]
```diff
@@ -9,18 +9,19 @@ import android.widget.TextView
 import androidx.core.content.ContextCompat
 import androidx.recyclerview.widget.RecyclerView
 
-internal class ListAdapter(
+class ListAdapter(
     private val context: Context,
```
Comment (gdragan) [inline app/src/main/java/com/example/filemanager/ListAdapter.kt]: I think you dont need to pass the context as argument, you can access it from view, please check line 27 `LayoutInflater.from(parent.context)` you are getting the context from parent `ViewGroup`, you can use that one and remove this one.

File: app/src/main/java/com/example/filemanager/ListAdapter.kt [hunk 5]
```diff
@@ -9,18 +9,19 @@ import android.widget.TextView
...
-    private val list: MutableList<ListModel>,
     private val itemClickListener: ItemClickListener
-) :
-    RecyclerView.Adapter<ListAdapter.ViewHolder>() {
+) : RecyclerView.Adapter<ListAdapter.ViewHolder>() {
 
-    internal inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    private val list = mutableListOf<ListModel>()
```
Comment (gdragan) [inline app/src/main/java/com/example/filemanager/ListAdapter.kt]: Can use a better name for this `val`, from it's name should be clear what data it holds in it, `files`, `filesList`.

File: app/src/main/java/com/example/filemanager/ListAdapter.kt [hunk 6]
```diff
@@ -29,13 +30,25 @@ internal class ListAdapter(
...
         val item = list[position]
-        holder.icon.setImageDrawable(ContextCompat.getDrawable(context, item.icon!!))
-        holder.fileName.text = item.fileName
-        holder.additionalInfo.text = item.additionalInfo
+
+        if (item.file.isDirectory) {
+            holder.icon.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.folder_icon))
+        } else if (item.file.isFile) {
+            holder.icon.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.file_icon))
+        }
```
Comment (gdragan) [inline app/src/main/java/com/example/filemanager/ListAdapter.kt]: Can simplify this `if else` block to remove duplicated code.

```
val fileIconRes = if(item.file.isDirectory) R.drawable.folder_icon else R.drawable.file_icon
holder.icon.setImageDrawable(ContextCompat.getDrawable(context, fileIconRes)
```

File: build.gradle [hunk 7]
```diff
@@ -5,7 +5,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:7.0.0"
+        classpath 'com.android.tools.build:gradle:7.0.1'
```
Comment (gdragan) [inline build.gradle:8]: Can extract this in a separate PR. Changing build conflig files has nothing to do with displaying files on UI. It is a good practice no to mix changes that does different things in the same PR.

File: app/src/main/java/com/example/filemanager/AllFilesViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+class AllFilesViewModel(
+    @SuppressLint("StaticFieldLeak")
+    private val context: Context
+) : ViewModel() {
+
+    val files = MutableLiveData<List<ListModel>>()
+
+    fun fetchFiles() {
+        val path = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            System.getenv("EXTERNAL_STORAGE")
```
Comment (gdragan) [inline app/src/main/java/com/example/filemanager/AllFilesViewModel.kt]: It doesnt look like an API provided by the Android SDK to load files. Can do some more investigations. There must be a better API to get the path to external storage directory.

**Review by gdragan — COMMENTED** [review]
Looks good. The title could be made shorter. It should not exceed 50 characters if possible. For example you could make it like:
`Display external storage files on portrait and landscape screens`

Also could use a more structured summary:
### Summary
- Added `AllFilesViewModel` responsible for loading external storage files
- The files are displayed on portrait and landscape screens
- In order to extract files and folders was used the `System.getenv("EXTERNAL_STORAGE")` API for versions Android R and above, otherwise `Environment.getExternalStorageDirectory().absolutePath` is used.

Below the summary can add some screenshots for reference:
....

**Review by gdragan — CHANGES_REQUESTED** [review]
...

File: app/src/main/java/com/example/filemanager/AllFilesViewModel.kt (hunk 2 above)
Comment (ecaterinagherbovei) [reply app/src/main/java/com/example/filemanager/AllFilesViewModel.kt]: I deleted context variable

File: app/src/main/java/com/example/filemanager/AllFilesViewModel.kt (hunk 3 above)
Comment (ecaterinagherbovei) [reply app/src/main/java/com/example/filemanager/AllFilesViewModel.kt]: I don't need them, I used logs to see if  methoda works

**Review by gdragan — APPROVED** [review]
